src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -92,12 +92,15 @@ private static IVecInt toVecInt(IntStream stream) {
 	public Map<String, ModCandidate> findCompatibleSet(Logger logger, Map<String, ModCandidateSet> modCandidateSetMap) throws ModResolutionException {
 		// First, map all ModCandidateSets to Set<ModCandidate>s.
 		boolean isAdvanced = false;
-		Map<String, Collection<ModCandidate>> modCandidateMap = new HashMap<>();
+		Map<String, List<ModCandidate>> modCandidateMap = new HashMap<>();
 		Set<String> mandatoryMods = new HashSet<>();
 
 		for (ModCandidateSet mcs : modCandidateSetMap.values()) {
 			Collection<ModCandidate> s = mcs.toSortedSet();
-			modCandidateMap.put(mcs.getModId(), s);
+			modCandidateMap.computeIfAbsent(mcs.getModId(), i -> new ArrayList<>()).addAll(s);
+			for (String modAlias : mcs.getModAliases()) {
+				modCandidateMap.computeIfAbsent(modAlias, i -> new ArrayList<>()).addAll(s);
+			}
```

Comment:
This appears to be broken when giving fabric loader an alias:

```json
{
  "schemaVersion": 1,
  "id": "fabricloader",
  "aliases": [
    "testmod"
  ]
```

Fails with a duplicate mod id of `fabricloader`.

Reply:
Doesn't seem to do the same for me?  I aliased fabricloader in the fabric.mod.json and it didn't give any errors.
I'd assume then that you have the testmod in your mods folder.

Reply:
Ill take another look, the testmod has the id of `test`

Reply:
In any case, it would probably be preferable for Loader to prefer an actual mod id to an alias which might be what the problem is here.

Reply:
Yes, I dont plan on using an alias in loader anytime soon, but the option should be there. Ill dig into it more to see whats going on.

Reply:
Seems to fail when going down the none advanced path (ModReslover:113). Make sure you dont have any mods in the mods dir when testing.

Reply:
I see what you mean, whoops